collectors/cgroups.plugin/cgroup-name.sh.in

```diff
@@ -53,18 +53,26 @@ function docker_get_name_classic() {
 }
 
 function docker_get_name_api() {
-	local id="${1}"
-	if [ ! -S "${DOCKER_HOST}" ]; then
-		warning "Can't find ${DOCKER_HOST}"
+	local path="/containers/${1}/json"
+	if [ -z "${DOCKER_HOST}" ]; then
+		warning "No DOCKER_HOST is set"
 		return 1
 	fi
 	if ! command -v jq >/dev/null 2>&1; then
 		warning "Can't find jq command line tool. jq is required for netdata to retrieve docker container name using ${DOCKER_HOST} API, falling back to docker ps"
 		return 1
 	fi
-
-	info "Running API command: /containers/${id}/json"
-	JSON=$(echo -e "GET /containers/${id}/json HTTP/1.0\\r\\n" | nc -U "${DOCKER_HOST}" | grep '^{.*')
+    if [ -S "${DOCKER_HOST}" ]; then
+        info "Running API command: curl --unix-socket ${DOCKER_HOST} http://localhost${path}"
+        JSON=$(curl --unix-socket ${DOCKER_HOST} http://localhost${path})
+    elif [ "${DOCKER_HOST}" == "/var/run/docker.sock" ]; then
```

Comment:
remove this else-if.
if DOCKER_HOST is not a socket, we just go for the API approach

Reply:
The reason I have this extra `elif` is because the default value is `/var/run/docker.sock`. If a user provides no env variables but also does not mount the socket, they will fall into this case and see some message.

I could make it say something else or direct them to the documentation (I'll rewrite and publish in a new diff) that explains how to provide access to Docker.